updating sql/jpa sample readmes with root password details

balopat opened the pull request:
Closes #376 
Closes #398

meltsufin commented on spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/README.adoc:
```diff
@@ -4,30 +4,35 @@ This sample application demonstrates how to use
+*[instance-connection-name]* with the instance connection name of the instance you created in
+step 1.
++
+Add the `spring.datasource.password` property with the password as the value.
```
Maybe we should put a placeholder for it there, commented out?

balopat replied:
I'll add the placeholder.
What do you think about adding it for username too?

meltsufin replied:
Sure!

joaoandremartins commented on spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/README.adoc:
```diff
@@ -4,30 +4,35 @@ This sample application demonstrates how to use
-+
-`$ gcloud auth application-default login`
+3. Still within the Google Cloud Console SQL page, create a new database in the instance from the
+"Databases" section. **We recommend you to setup a root password** for your instance (in PostgreSQL you have to).
```
I have reservations asking users to set a password on their instance.
With the Cloud SQL tunnel + OAuth2 authentication and authorization, the password is actually not necessary and if I had a recommendation, it would actually recommend against setting a password.

balopat replied:
With PostgreSQL you have to set it, there is no way out. 
Setting passwords is just simply best practice in general. So I would still go with passwords by default. 
However, we can add a footnote, that if users want to try out for development purposes, there is a "passwordless" option for MySQL.

joaoandremartins commented on spring-cloud-gcp-samples/spring-cloud-gcp-sql-sample/README.adoc:
```diff
@@ -15,15 +15,15 @@ Cloud Project with billing enabled.
 
 3. Still within the Google Cloud Console SQL page, create a new database in the instance from the
-"Databases" section.
+"Databases" section. **We recommend you to setup a root password** for your instance (in PostgreSQL you have to).
```
Ditto on recommending users to set a password.

artembilan commented on spring-cloud-gcp-samples/spring-cloud-gcp-sql-sample/README.adoc:
```diff
@@ -1,12 +1,12 @@
-= Spring Cloud GCP SQL Starter Example
+= Spring Cloud GCP SQL Code Sample
```
I think we don't need `Code` word here.

balopat commented:
@artembilan: fixed
@joaoandremartins: added footnote (see it [in action](https://github.com/spring-cloud/spring-cloud-gcp/tree/sql_samples_root_password_instructions/spring-cloud-gcp-samples/spring-cloud-gcp-sql-sample)) 
@meltsufin: placeholders added

PTAL, thank you!

joaoandremartins commented on spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/README.adoc:
```diff
@@ -1,33 +1,38 @@
-+
-`$ gcloud auth application-default login`
+3. Still within the Google Cloud Console SQL page, create a new database in the instance from the
+"Databases" section. **We recommend you to setup a root password** for your instance footnoteref:[passwordless, if you are using MySQL, you can actually create the database instance without root password!]
```
Sorry to be beating on a dead horse, but I would rather we didn't "recommend" people to create a root password. It should be safe enough to rely on the OAuth2 authentication and it makes config simpler to not have one there.
If the password is mandatory for PostgreSQL, how about we only add a parenthesis saying something like "PostgreSQL requires a password. Please set one and make note of it"?

joaoandremartins commented on spring-cloud-gcp-samples/spring-cloud-gcp-sql-sample/README.adoc:
```diff
@@ -15,15 +15,15 @@ Cloud Project with billing enabled.
 
 3. Still within the Google Cloud Console SQL page, create a new database in the instance from the
-"Databases" section.
+"Databases" section. **We recommend you to setup a root password** for your instance footnoteref:[passwordless, If you are using MySQL, you can actually create the database instance without root password!]
```
Ditto for the "recommended".

meltsufin commented on spring-cloud-gcp-samples/spring-cloud-gcp-sql-sample/src/main/resources/application.properties:
```diff
@@ -6,5 +6,11 @@ spring.datasource.continue-on-error=true
+# Leave empty for root, uncomment and fill out if you specified a user
+#spring.datasource.username=
+
+# Uncomment if root password is specified (recommended)
```
Do you want to keep this "recommended"?

balopat replied:
great spot! thanks :)

meltsufin commented on spring-cloud-gcp-samples/spring-cloud-gcp-data-jpa-sample/src/main/resources/application.properties:
```diff
@@ -6,4 +6,10 @@ spring.jpa.hibernate.ddl-auto=create-drop
+#spring.datasource.username=
+
+# Uncomment if root password is specified (recommended)
+#spring.datasource.password=
```
And here.
